app/services/container_project_service.rb

```diff
@@ -0,0 +1,51 @@
+class ContainerProjectService
```

Comment:
the file name is somewhat misleading... you can think it's a relationship between project to the service entity